Elastica/EventSubscriber/UpdateNodeIndexedSubscriber.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace OpenOrchestra\Elastica\EventSubscriber;
+
+use OpenOrchestra\Elastica\Indexor\NodeIndexor;
+use OpenOrchestra\ModelInterface\Event\NodeEvent;
+use OpenOrchestra\ModelInterface\Model\ReadNodeInterface;
+use OpenOrchestra\ModelInterface\NodeEvents;
+use OpenOrchestra\ModelInterface\Repository\ReadNodeRepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use OpenOrchestra\Elastica\Exception\IndexorWrongParameterException;
+
+/**
+ * Class UpdateNodeIndexedSubscriber
+ */
+class UpdateNodeIndexedSubscriber implements EventSubscriberInterface
+{
+    protected $nodeRepository;
+    protected $nodeIndexor;
+
+    /**
+     * @param ReadNodeRepositoryInterface $nodeRepository
+     * @param NodeIndexor                 $nodeIndexor
+     */
+    public function __construct(ReadNodeRepositoryInterface $nodeRepository, NodeIndexor $nodeIndexor)
+    {
+        $this->nodeRepository = $nodeRepository;
+        $this->nodeIndexor = $nodeIndexor;
+    }
+
+    /**
+     * @param NodeEvent $event
+     *
+     * @throws IndexorWrongParameterException
+     */
+    public function updateIndexedNode(NodeEvent $event)
+    {
+        $node = $event->getNode();
+        $lastPublishedNode = $this->nodeRepository->findOneCurrentlyPublished($node->getNodeId(), $node->getLanguage(), $node->getSiteId());
+        if ($lastPublishedNode instanceof ReadNodeInterface) {
```

Comment:
@alavieille : You could optimize the process by also checking that the $node->getId() is equal to the $lastPublishedNode->getId(). If not, that means that the index process is not required (only if the publishing process involves this method too)